app/controllers/api/trainees/withdraws_controller.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module Api
+  module Trainees
+    class WithdrawsController < Api::BaseController
+      def create
+        if trainee.blank?
+          render(json: { error: "Trainee not found" }, status: :not_found)
+        else
+          withdraw_trainee if withdraw_allowed?
+
+          render(json: { status: "Trainee withdrawal request accepted", trainee: TraineeSerializer.new(trainee).as_json }, status: :accepted)
+        end
+      end
+
+    private
+
+      def withdraw_trainee
+        trainee.update(withdrawal_params)
+        trainee.withdraw!
+        ::Trainees::Withdraw.call(trainee:)
+      end
+
+      def trainee
+        @trainee ||= Trainee.find_by(slug:)
```

Comment:
This seems dangerous to me. We should really have a scoped lookup, something like `current_provider.trainees.find_by(slug:)`. Feels like this belongs in it's own little helper method.

Reply:
Yep that was my stop gap whilst the other streams of work was getting there

Reply:
rebased and fixed now